templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java

```diff
@@ -76,18 +76,18 @@ public AzureTREAuthenticatedUser authenticateUser(final Credentials credentials)
 
         // Getting headers from the oauth2 proxy
         final String accessToken = credentials.getRequest().getHeader("X-Forwarded-Access-Token");
-        final String prefEmail = credentials.getRequest().getHeader("X-Forwarded-Email");
+        final String prefUsername = credentials.getRequest().getHeader("X-Forwarded-Preferred-Username");
 
         if (Strings.isNullOrEmpty(accessToken)) {
             LOGGER.error("access token was not provided");
             return null;
         }
-        if (Strings.isNullOrEmpty(prefEmail)) {
-            LOGGER.error("email was not provided");
+        if (Strings.isNullOrEmpty(prefUsername)) {
+            LOGGER.error("username was not provided");
```

Comment:
This error message might be misleading during troubleshooting. The value that was not provided was Preferred Username, not Username.